protocol-specs/generators/proof_of_misbehavior_transaction/index.js

```diff
@@ -0,0 +1,531 @@
+
+	const payloadHashBuffer = hexToBuffer(block.payloadHash);
+
+	const generatorPublicKeyBuffer = hexToBuffer(block.generatorPublicKey);
+
+	const blockSignatureBuffer = block.blockSignature
+		? hexToBuffer(block.blockSignature)
+		: Buffer.alloc(0);
+
+	return Buffer.concat([
+		blockVersionBuffer,
+		timestampBuffer,
+		previousBlockBuffer,
+		seedRevealBuffer,
+		heightBuffer,
+		maxHeightPreviouslyForgedBuffer,
+		maxHeightPrevotedBuffer,
+		numTransactionsBuffer,
+		totalAmountBuffer,
+		totalFeeBuffer,
+		rewardBuffer,
+		payloadLengthBuffer,
+		payloadHashBuffer,
+		generatorPublicKeyBuffer,
+		blockSignatureBuffer,
+	]);
+};
+
+const sign = (block, privateKey) =>
+	signDataWithPrivateKey(hash(getBytes(block)), Buffer.from(privateKey, 'hex'));
+
+const getId = transactionBytes => {
+	const transactionHash = hash(transactionBytes);
+	const bufferFromFirstEntriesReversed = getFirstEightBytesReversed(
+		transactionHash,
+	);
+	const transactionId = bufferToIntAsString(bufferFromFirstEntriesReversed);
+
+	return transactionId;
+};
+
+const assetToBytes = tx => {
+	const { header1, header2 } = tx.asset;
+
+	const assetBuffer = Buffer.concat([getBytes(header1), getBytes(header2)]);
+
+	return assetBuffer;
+};
+
+const serialize = tx => {
+	const transactionNonce = intToBuffer(tx.nonce.toString(), 8);
+
+	const buf = Buffer.concat([
+		Buffer.alloc(1, tx.type),
+		transactionNonce,
+		hexToBuffer(tx.senderPublicKey),
+		intToBuffer(tx.fee.toString(), 8),
+		assetToBytes(tx),
+	]);
+
+	return buf;
+};
+
+const createSignatureObject = (txBuffer, account) => ({
+	signature: signData(
+		hash(Buffer.concat([hexToBuffer(networkIdentifier), txBuffer])),
+		account.passphrase,
+	),
+});
+
+const accounts = {
+	reporter: {
+		address: '16313739661670634666L',
+		publicKey:
+			'c094ebee7ec0c50ebee32918655e089f6e1a604b83bcaa760293c61e0f18ab6f',
+		passphrase:
+			'wagon stock borrow episode laundry kitten salute link globe zero feed marble',
+		balance: '10000000000000000',
+		encryptedPassphrase:
+			'iterations=1&salt=e8c7dae4c893e458e0ebb8bff9a36d84&cipherText=c0fab123d83c386ffacef9a171b6e0e0e9d913e58b7972df8e5ef358afbc65f99c9a2b6fe7716f708166ed72f59f007d2f96a91f48f0428dd51d7c9962e0c6a5fc27ca0722038f1f2cf16333&iv=1a2206e426c714091b7e48f6&tag=3a9d9f9f9a92c9a58296b8df64820c15&version=1',
+		password: 'elephant tree paris dragon chair galaxy',
+	},
+	forger: {
+		address: '10881167371402274308L',
+		publicKey:
+			'addb0e15a44b0fdc6ff291be28d8c98f5551d0cd9218d749e30ddb87c6e31ca9',
+		passphrase:
+			'actress route auction pudding shiver crater forum liquid blouse imitate seven front',
+		balance: '0',
+		delegateName: 'genesis_100',
+	},
+};
+
+const forgerKeyPair = getPrivateAndPublicKeyBytesFromPassphrase(
+	accounts.forger.passphrase,
+);
+
+/*
+	Scenario 1:
+
+	b1.maxHeightPrevoted==b2.maxHeightPrevoted &&  b1.height>=b2.height
+*/
+
+const scenario1Header1 = {
+	version: 1,
+	timestamp: 1,
+	previousBlockId: 1,
+	seedReveal: '1',
+	height: 300000,
```

Comment:
I think in this case `header1` will be b2, so the height should be opposite

Reply:
`header1` is b1 since it has a larger height.

Reply:
let scenario1Header1 b1 and scenario1Header2, and 
`(b1.maxHeightPreviouslyForged==b2.maxHeightPreviouslyForged && b1.maxHeightPrevoted==b2.maxHeightPrevoted && b1.height>b2.height)`
when this condition applies, `b1` and `b2` switches

https://github.com/LiskHQ/lips/blob/master/proposals/lip-0014.md#detecting-contradicting-block-headers